stable-diffusion/model.py

```diff
@@ -0,0 +1,295 @@
+        """Generate images from noise using the backward diffusion process.
+            
+        Args:
+            prompt (str or List[str]) — The prompt or prompts to guide the image generation.
+            height (int, optional, defaults to self.unet.config.sample_size * self.vae_scale_factor) — The height in pixels of the generated image.
+            width (int, optional, defaults to self.unet.config.sample_size * self.vae_scale_factor) — The width in pixels of the generated image.
+            num_inference_steps (int, optional, defaults to 50) — The number of denoising steps. More denoising steps usually lead to a higher quality image at the expense of slower inference.
+            guidance_scale (float, optional, defaults to 7.5) — Guidance scale as defined in Classifier-Free Diffusion Guidance. guidance_scale is defined as w of equation 2. of Imagen Paper. Guidance scale is enabled by setting guidance_scale > 1. Higher guidance scale encourages to generate images that are closely linked to the text prompt, usually at the expense of lower image quality.
+            negative_prompt (str or List[str], optional) — The prompt or prompts not to guide the image generation. Ignored when not using guidance (i.e., ignored if guidance_scale is less than 1). Must be the same length as list of prompts.
+            num_images_per_prompt (int, optional, defaults to 1) — The number of images to generate per prompt.
+        """
+        num_images_per_prompt = num_images_per_prompt if num_images_per_prompt else self.num_images_per_prompt
+        batch_size = 1 if isinstance(prompt, str) else len(prompt)
+
+        if negative_prompt:
+            negative_prompt_bs = 1 if isinstance(negative_prompt,
+                                                 str) else len(negative_prompt)
+            if negative_prompt_bs != batch_size:
+                raise ValueError(
+                    f'len(prompts) and len(negative_prompts) must be the same. A negative prompt must be provided for each given prompt.'
+                )
+
+        vae_scale_factor = 8
+        height = height or self.unet.config.sample_size * vae_scale_factor
+        width = width or self.unet.config.sample_size * vae_scale_factor
+
+        device = self.vae.device
+        # tokenize and encode text prompt
+        text_input = self.tokenizer(prompt,
+                                    padding="max_length",
+                                    max_length=self.tokenizer.model_max_length,
+                                    truncation=True,
+                                    return_tensors="pt")
+        text_embeddings = self.text_encoder(text_input.input_ids.to(device))[0]
+
+        # classifier free guidance + negative prompts
+        # negative prompt is given in place of the unconditional input in classifier free guidance
+        unconditional_input = [
+            ""
+        ] * batch_size if not negative_prompt else negative_prompt
```

Comment:
mostly trying to get this on one line instead of three
```suggestion
        unconditional_input = negative_prompt or ([""] * batch_size)
```